pkg/registry/generic/etcd/etcd.go

```diff
@@ -295,22 +307,31 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 		}
 
 		creating = false
-		newVersion, err := e.Helper.Versioner.ObjectResourceVersion(obj)
-		if err != nil {
-			return nil, nil, err
-		}
-		if newVersion != version {
-			// TODO: return the most recent version to a client?
-			return nil, nil, kubeerr.NewConflict(e.EndpointName, name, fmt.Errorf("the object has been modified; please apply your changes to the latest version and try again"))
+		if useLatestResourceVersion {
+			// Update the object's resource version to match the latest etcd object's resource version.
+			err = e.Helper.Versioner.UpdateObject(obj, node)
```

Comment:
Re: using version directly, @bgrant0607 suggested to call the whole UpdateObject rather than factor out a UpdateResourceVersion out of it and call just that.
He said that in future, if we add any other field to UpdateObject, we want it to be updated here as well.